### ISSUE TITLE
populating a nested subdoc of an array causes the other fields of the array elements to not be brought back

### ISSUE THREAD

**Issue by Piper0804** [Automattic/mongoose#5701]
This behavior was introduced as of mongoose v4.11.14.

I don't have a working script to share, but the gist of the issue is that when populating a nested document which is part of an array, the sub fields of the array elements are no longer retrieved. This is not the behavior in previous versions of mongoose, where the populated data would come back as well as the other selected sub fields of the array element.

my schema :
```
var schema = new mongoose.Schema({
	...
	'an_object' : [{
		_id : false,
		'val_one' : { type : String, required : true },
		'val_two' : { type : Number, required : true }
		'ref_one' : { ref : 'other_type', type : dbType.ObjectId, required : true },
	}]
});
```

call to find:
```
var populate_options = [{ path : 'an_object.ref_one', select : { 'a_val' : 1, 'another_val' : 1 }}];

_find({}, 'an_object').populate(populate_options).exec( function (error, items) {
	...
```
Since I'm selecting 'an_object' the expectation is that all fields within an_object are returned. (They are when populate is not used).

other info :  (mongodb v3.4, node v6.9.1)

**Comment by jsl9208** [comment]
same here, mongoose v4.12.0

**Comment by welkinwong** [comment]
same here

**Comment by vkarpov15** [comment]
I suspect this is a duplicate of #5669 . Can you upgrade to 4.12.2 and see if you still get this issue?

**Comment by Piper0804** [comment]
The issue began as of 4.11.14, which introduced fix 5669. It appears that
5669 caused this new issue.

**Comment by Piper0804** [comment]
I just saw that 5669 was addressed again in 4.12.2. I can't confirm yet if
it's fixed, will try it this coming week.

**Comment by vkarpov15** [comment]
As far as I can tell, this is fixed by #5669 in 4.12.2. Please re-open with detailed repro instructions if you're still experiencing this issue.

**Comment by Piper0804** [comment]
I confirmed it's fixed for me while using 4.12.3. Thanks!
